users/mnil/mnil.c

```diff
@@ -0,0 +1,133 @@
+#include QMK_KEYBOARD_H
+#include "config.h"
+#include "mnil.h"
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case M_TILD:  // ~
+            if (record->event.pressed) {
+                tap_code16(RALT(KC_RBRC));
+                tap_code(KC_SPC);
+            } else {
+            }
+            break;
+        case M_CIRC:  // ^
+            if (record->event.pressed) {
+                tap_code16(S(KC_RBRC));
+                tap_code(KC_SPC);
+            } else {
+            }
+            break;
+        case M_BTCK:  // `
+            if (record->event.pressed) {
+                tap_code16(S(KC_EQL));
+                tap_code(KC_SPC);
+            } else {
+            }
+            break;
+        case QWE_COL:  // Swap default keymap layer
+            if (record->event.pressed) {
+                layer_invert(_COLEMAK);
+                layer_invert(_QWERTY);
```

Comment:
I think you actually want to use the default layer rather than this.  

`DF(x)` can do this, or you can use `set_single_persistent_default_layer(layer)` to set this persistantly (eg, will stay after power loss/reset)

Reply:
This is only used in my planck board, and it seems it doesn't have persistent storage.
https://github.com/qmk/qmk_firmware/issues/8535
It also works if I remove: `layer_invert(_COLEMAK);`
I don't really care about persistent storage anyway, so if you have a more specific suggestion on how to add `DF(x)` using only one key to change between colemak and qwerty I am all ears.

Reply:
`DF(x)` as the keycode will work.  But it won't be persistant (in wouldn't normally be, either).  

But for reference, the reason I recommend `DF` instead of this is: 
https://docs.qmk.fm/#/keymap?id=keymap-and-layers